R/utility.R

```diff
@@ -179,3 +179,28 @@ getJobResult <- function(jobId = "", ...){
 
   return(results)
 }
+
+validateClusterConfig <- function(clusterFilePath){
+  if (file.exists(clusterFilePath)) {
+    pool <- rjson::fromJSON(file = clusterFilePath)
+  }
+  else{
+    pool <- rjson::fromJSON(file = file.path(getwd(), clusterFilePath))
+  }
+
+  stopifnot(!is.null(pool$poolSize))
+  stopifnot(!is.null(pool$poolSize$dedicatedNodes))
+  stopifnot(!is.null(pool$poolSize$lowPriorityNodes))
+
+  stopifnot(is.character(pool$name))
+  stopifnot(is.character(pool$vmSize))
+  stopifnot(is.character(pool$poolSize$autoscaleFormula))
+
+  stopifnot(is.double(pool$poolSize$dedicatedNodes$min))
+  stopifnot(is.double(pool$poolSize$dedicatedNodes$max))
+  stopifnot(is.double(pool$poolSize$lowPriorityNodes$min))
+  stopifnot(is.double(pool$poolSize$lowPriorityNodes$max))
+  stopifnot(is.double(pool$maxTasksPerNode))
+
+  TRUE
```

Comment:
I've added entries for min/max for both low pri and dedicated. I also added some validations that will make debugging harder for the user such as bad autoscale formula.